packages/react-native/ReactAndroid/src/main/java/com/facebook/react/runtime/ReactLifecycleStateManager.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.react.runtime
+
+import android.app.Activity
+import androidx.annotation.Nullable
+import com.facebook.infer.annotation.Nullsafe
+import com.facebook.infer.annotation.ThreadConfined
+import com.facebook.infer.annotation.ThreadConfined.UI
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.common.LifecycleState
+
+@Nullsafe(Nullsafe.Mode.LOCAL)
+internal class ReactLifecycleStateManager(
+        private val bridgelessReactStateTracker: BridgelessReactStateTracker
+) {
+  private var state: LifecycleState = LifecycleState.BEFORE_CREATE
+
+  fun getLifecycleState(): LifecycleState = state
```

Comment:
Could you try exposing this as:

```suggestion
  val lifecycleState: LifecycleState
    get() = state
```

A bit more Kotlin idiomatic.

Reply:
made the suggested changes